CRM/Member/Form.php

```diff
@@ -223,4 +223,50 @@ public function storeContactFields($formValues) {
     }
   }
 
+  /**
+   * Create a recurring contribution record.
+   *
+   * Recurring contribution parameters are set explicitly rather than merging paymentParams because it's hard
+   * to know the downstream impacts if we keep passing around the same array.
+   *
+   * @param $paymentParams
+   *
+   * @return array
+   * @throws \CiviCRM_API3_Exception
+   */
+  protected function processRecurringContribution($paymentParams) {
+    $membershipID = $paymentParams['membership_type_id'][1];
+    $contributionRecurParams = array(
+      'contact_id' => $paymentParams['contactID'],
+      'amount' => $paymentParams['total_amount'],
+      'payment_processor_id' => $paymentParams['payment_processor_id'],
+      'campaign_id' => $paymentParams['campaign_id'],
+      'financial_type_id' => $paymentParams['financial_type_id'],
+      'is_email_receipt' => $paymentParams['is_email_receipt'],
+      // This is not great as it could also be direct debit - but is consistent with elsewhere & all need fixing.
+      'payment_instrument_id' => 1,
```

Comment:
I get these bunch of warnings (may be use CRM_Utils_Array::value('key', $array) instead)
    Notice: Undefined index: campaign_id in CRM_Member_Form->processRecurringContribution() (line 243 of /Users/yashodha/git/civicrm/CRM/Member/Form.php).
    Notice: Undefined index: is_email_receipt in CRM_Member_Form->processRecurringContribution() (line 245 of /Users/yashodha/git/civicrm/CRM/Member/Form.php).
    Notice: Undefined index: invoiceID in CRM_Member_Form->processRecurringContribution() (line 248 of /Users/yashodha/git/civicrm/CRM/Member/Form.php).